src/v/cluster/scheduling/allocation_node.h

```diff
@@ -55,6 +55,10 @@ class allocation_node {
         // state is being updated
         return _max_capacity - std::min(_allocated_partitions, _max_capacity);
     }
+    allocation_capacity priority_partition_capacity() const {
```

Comment:
I think this new function and the old one `partition_capacity` need clear comments, because at least to me the new semantics aren't clear here. If _max_capacity is 100 and we have 90 non priority partitions allocated and 0 priority ones, the priority capacity is still 100? That would put us over _max_capacity...